### PR TITLE
gcp overlay

### PR DESCRIPTION
**Pull request by honnix**
This roughly reflects how we deployed flyte to GKE. Please note that this is by no means to set up a proper production grade flyte.

A few things to note:

* Most things are copied from `sandbox` with modification
* cloudsql proxy is used and the password is stored in k8s secret named as `db-user-pass`
* GCS is configured using `stow`
* Workload identify on GKE is beta so we don't use it yet; flyte is operating under the service account assigned to the GKE cluster
* For ingress, GCP internal load balancer is used; prometheus endpoints (admin, propeller, and datacatalog) are exposed via load balancer, so the scraper is ready to go
* Viewing input/output on flyte console does not work due to GCS path handling

### CODE REVIEW

**Comment by honnix** [comment]
@kumare3 PTAL, thanks.

**Comment by kumare3** [comment]
I have taken a look and this is great, want to see how we merge this. As this is not directly usable right, so we need to give guidance as to how to use it.

**Comment by honnix** [comment]
Yeah. I will find some time very soon to add some readme alike things.

**Comment by honnix** [comment]
@kumare3 I added a readme describing how to further configure gcp overlay. PTAL again. BTW e2e test has stopped working due to timeout waiting for k8s to start.

**Comment by honnix** [comment]
Debugging e2e test in https://github.com/lyft/flyte/pull/273 but I couldn't get dockernetes started properly. More specifically kube-proxy won't start.

**Comment by honnix** [comment]
Ping @kumare3

File: kustomize/overlays/gcp/admin/cron.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: syncresources
+  namespace: flyte
+spec:
+  schedule: "*/1 * * * *"
+  jobTemplate:
+    spec:
+      template:
```
Comment (kumare3) [inline kustomize/overlays/gcp/admin/cron.yaml:10]: we need a serviceaccount here

**Comment by kumare3** [comment]
@honnix can we merge this?

**Comment by kumare3** [comment]
This looks good to me, once you merge, lets ship it!

**Comment by honnix** [comment]
@kumare3 Sounds good to me. I will take a final check today. Thanks for fixing this PR.

**Comment by kumare3** [comment]
Thank you this is awesome

**Comment by honnix** [comment]
@kumare3 I upgraded some of the image versions and did some cleanup as well. PTAL.

**Comment by kumare3** [comment]
Looks great to me

**Pull request merged**